### ISSUE TITLE
training tricks of HIST

### ISSUE THREAD

**Issue by BeckyYu96** [microsoft/qlib#1136]
Hi there,

Thanks for making the codes of HIST public. I may need some discussion on two topics.

1. `iter_daily_shuffle` or `iter_daily`
Regarding the order of training samples fed to model, I've noticed that the days were shuffled by default `(`iter_daily_shuffle`).` And when I changed the setting to train samples day by day `(iter_daily),` I met a huge performance decrease.  I'm a little confused about that. Is the method `iter_daily_shuffle` on suspicion of some information leakage since the model see later samples at first? Have you met the same situation when training the model?

2. split of train/valid/test sets
It introduces great randomness to model performance when I split train/valid/test sets in different time windows. I'm not sure if there is some workaround to overcome this kind of randomness and make the model more robust.

Any idea or help would be appreciated!

**Comment by you-n-g** [comment]
Hi, @BeckyYu96

1.  Reference from https://datascience.stackexchange.com/a/24524
> Shuffling data serves the purpose of reducing variance and making sure that models remain general and overfit less. 

2. You can do it by rolling training like [this](https://tsfresh.readthedocs.io/en/latest/text/forecasting.html)
